### PR TITLE
feat: add "context" prop to Outlet

### PR DESCRIPTION
**Pull request by mcansh**
accidentally originally opened it against [remix](https://github.com/remix-run/remix/pull/939)

### CODE REVIEW

**Review by kentcdodds — COMMENTED** [review]
LGTM. I'll add some docs 👍

**Review by kentcdodds — APPROVED** [review]
LGTM 👍

File: docs/api.md [hunk 1]
```diff
@@ -545,6 +550,53 @@ function App() {
...
+      <h1>Dashboard</h1>
+      <Outlet context={user} />
+    </div>
+  );
+}
+
+export function useUser() {
+  return useOutletContext<ContextType>();
+}
+```
```
Comment (ryanflorence) [inline docs/api.md:584]: Let's remove the wrapper here and just do:

```jsx
const { user } = useOutletContext();
```

Also ditch the typescript in the doc, TS users can figure it out, or add a second section in this doc showing the generic usage.

Comment (kentcdodds) [reply docs/api.md:605]: I've adjusted things to be better 👍

**Pull request merged**